integrations/access/opsgenie/client.go

```diff
@@ -153,20 +157,59 @@ func (og Client) CreateAlert(ctx context.Context, reqID string, reqData RequestD
 	if resp.IsError() {
 		return OpsgenieData{}, errWrapper(resp.StatusCode(), string(resp.Body()))
 	}
+
+	logger.Get(ctx).Debugf("Create Alert Response: %+v", result)
+	alertRequestResult, err := og.tryGetAlertRequestResult(ctx, result.RequestID)
```

Comment:
So what happens if this fails for some reason?

I wonder if it would be more robust to move this logic to request resolution time. So we would store `RequestID` here, and then the plugin would attempt to query `AlertID` when it needs to resolve the incident. WDYT?

Reply:
Have been trying out implementing this this morning and I think it should work but it gets a bit messy now having to double check everywhere we interact with an alert whether or not we have the alert or the request id and resolving it if appropriate. I think we might end up having to change some of the shared plugin code as well if we go this route to allow the accessrequest/MessageData stuff to represent a 'message' that we dont have the ID of yet

Reply:
In the event this fails it could go one of two ways depending on if the request ever does end up completing successfully after the retries fail. 
- 1: The alert is never created and that is why the retries failed - in this case the plugin just never interacts with that access request again.
- 2: The alert is eventually created after the retries all fail  - in this case the alert is created but the plugin wont be able to send updates or resolve it

Reply:
The auto-approval will still work in both of these cases, right? As long as we're able to determine if requester is on-call. Cause we never interact with alert/request ID to actually approve Teleport's request - just the alert will not be closed in Opsgenie, correct?

Reply:
Yes, the notify error in opsgenie/app.go and the error from trying to resolve the request are put into an aggregate error instead of returning early.

Reply:
Ok, in that case, let's do this:

- Add a comment here explaining that if this fails, Teleport request approval and auto-approval will still work but incident in Opsgenie won't be auto-closed.
- Increase the timeout to something more than 30s, maybe a couple minutes?

Reply:
Added in the comment and upped the timeout, does this look okay now?

Reply:
Yeah I already approved so let's get this merged and do the backports.